### PR TITLE
Fix FindSFML.cmake can't find SFML 2.1

### PR DESCRIPTION
**Pull request by xelez**
After commit https://github.com/SFML/SFML/commit/1c46ec7c37702a01e8397d63c3e3a0b9e70ff87b that updates version to 2.2, FindSFML.cmake can't find SFML 2.1 (and maybe other versions too).

That happens because Config.hpp of 2.1 hasn't got define for SFML_VERSION_PATCH. So regexp in FindSFML.cmake finds nothing.

I deleted first REGEX MATCH, because it seems to be just a little optimization, that gets things more complicated.

### CODE REVIEW

**Comment by mantognini** [comment]
Works fine for me. :+1: 

I've added it to 2.3.2 task list.

File: cmake/Modules/FindSFML.cmake [hunk 1]
```diff
@@ -89,10 +89,12 @@ if(SFML_FIND_VERSION AND SFML_INCLUDE_DIR)
         set(SFML_CONFIG_HPP_INPUT "${SFML_INCLUDE_DIR}/SFML/Config.hpp")
     endif()
     FILE(READ "${SFML_CONFIG_HPP_INPUT}" SFML_CONFIG_HPP_CONTENTS)
-    STRING(REGEX MATCH ".*#define SFML_VERSION_MAJOR ([0-9]+).*#define SFML_VERSION_MINOR ([0-9]+).*#define SFML_VERSION_PATCH ([0-9]+).*" SFML_CONFIG_HPP_CONTENTS "${SFML_CONFIG_HPP_CONTENTS}")
     STRING(REGEX REPLACE ".*#define SFML_VERSION_MAJOR ([0-9]+).*" "\\1" SFML_VERSION_MAJOR "${SFML_CONFIG_HPP_CONTENTS}")
     STRING(REGEX REPLACE ".*#define SFML_VERSION_MINOR ([0-9]+).*" "\\1" SFML_VERSION_MINOR "${SFML_CONFIG_HPP_CONTENTS}")
     STRING(REGEX REPLACE ".*#define SFML_VERSION_PATCH ([0-9]+).*" "\\1" SFML_VERSION_PATCH "${SFML_CONFIG_HPP_CONTENTS}")
+    if (NOT MATCHES)
```
Comment (eXpl0it3r) [inline cmake/Modules/FindSFML.cmake]: What does this evaluate?

Comment (mantognini) [inline cmake/Modules/FindSFML.cmake]: Oh, good catch! I though it was a nice trick of CMake but it's not. If I bump SFML version to 2.3.1 this script will always set `SFML_VERSION_PATCH` to 0.

Comment (mantognini) [inline cmake/Modules/FindSFML.cmake]: Should be something like `if (NOT "${SFML_VERSION_PATCH}" MATCHES "^[0-9]+$")` instead.

Comment (Bromeon) [inline cmake/Modules/FindSFML.cmake]: @xelez Could you change this accordingly? Would be nice if SFML were found again :wink:

**Comment by binary1248** [comment]
Bump.

**Comment by eXpl0it3r** [comment]
@xelez Can you update the PR? Otherwise we'll have to create a new PR ourselves. :wink:

**Comment by xelez** [comment]
Fixed, checked it on 2.0, 2.1, 2.2, 2.3, 2.3.1, hope I've missed nothing this time.

P.S. Oh, em.. sorry for long waiting, I'm quite ashamed both for the response time and that stupid mistake. =(

**Comment by eXpl0it3r** [comment]
Thanks! No need to feel bad.

Could you squash the two commits into one?

**Comment by xelez** [comment]
@eXpl0it3r like that?

**Comment by eXpl0it3r** [comment]
Yes!

This PR has been added to my merge list, meaning it will be merged soon into the branch 2.3.x, unless someone raises any concerns.

**Comment by eXpl0it3r** [comment]
Now that I test it, I wonder what exactly we "gain" from this, because if you try to use for example SFML 2.1 with the FindSFML.cmake from this PR, it may find SFML, but it will still search for OGG, Vorbis, FLAC, etc. even though they are not used with SFML 2.1...

**Comment by xelez** [comment]
Well, if someone uses SFML and cmake in their project, I think what they usually do is just copy this file to their Modules directory (I don't use SFML by myself, maybe some day =)). And now when someone wants to build that project against default libraries on, for example, Ubuntu they see quite strange error about can't find SFML and think "Eeerm.. what? Here it is! Why can't it find it?". That's how I found this bug. And I thought that after I fixed it in my case I should share the solution.

Hmm... on the other hand... if someone uses SFML 2.2 and uses current FindSFML... maybe it's not quite clever to build against previous version of the library, even if it's said it's supported.

And in about a year or probably less most of the people will use and support only SFML 2.2 or later, so if there is "gain", then it's probably very small.

Oh, and another thought. If current version of FindSFML fails on 2.1 or less.. maybe a better approach is to cleanup this file and make it clear, that it supports only 2.2 or later?

By the way, here http://www.sfml-dev.org/tutorials/2.3/start-linux.php it's suggested to apt-get install libsfml-dev... and it's 2.1 in ubuntu 14.04 for example.

**Comment by eXpl0it3r** [comment]
If nobody has anything against it, I'll merge this, even if it only fixes things partially - FindSFML.cmake will still search for the wrong dependencies when choosing SFML 2.0/2.1/2.2.

**Comment by TankOs** [comment]
It won't hurt, but a new issue has to be created that establishes backwards compatibility.

**Comment by eXpl0it3r** [comment]
Looking at the code again, it still seems to be wrong. Before we tried to match the major, minor and patch version in the config file. Now it just replaces things and afterwards checks if the patch version is set or not. Meaning we do not actually check first if the #defines even exists.

**Comment by eXpl0it3r** [comment]
Can anyone else confirm that this code is good (or bad)?

**Comment by TankOs** [comment]
If it doesn't help in the end, then it's of no big use. It can be applied, but like said, to properly support older versions, we will have to add branching to FindSFML.cmake to cover different dependencies etc.

**Comment by eXpl0it3r** [comment]
No I mean does the current code actually work. I mean the first statement has been completely removed and note replaced with anything else. So I'm just wondering.

**Comment by binary1248** [comment]
The first line was just a filter making sure all version defines were present. Removing it removes the check.

**Comment by eXpl0it3r** [comment]
> Removing it removes the check.

And is that okay or not?

**Comment by binary1248** [comment]
Well... it works, but it would allow bogus config files without a major or minor version also slip through. :stuck_out_tongue:

**Comment by eXpl0it3r** [comment]
Well if nobody has anything against this PR and it's implications as see above in the next days I'll finally merge it.

**Comment by TankOs** [comment]
I won't repeat myself again. :D :+1:

**Comment by eXpl0it3r** [comment]
Merged in 110feeb7f2f78f4b8f1b51dc1b61ae1b0aece9ce on branch 2.3.x.

Also created #950 so we can keep track of the issue and implement a "proper" fix at a later point.

**Comment by TankOs** [comment]
Thanks!

**Pull request closed without merging**